### PR TITLE
Fill match arms assist: add remaining arms for tuple of enums

### PR DESCRIPTION
**Pull request by iDawer**
Fix for #8493

However, the assist is still flaky and does not use `hir_ty::diagnostics::match_check`

### CODE REVIEW

File: crates/ide_assists/src/handlers/fill_match_arms.rs [hunk 1]
```diff
@@ -481,6 +492,67 @@ fn main() {
...
+                match (a, b) {
+                    (A::Two, B::One) => {}
+                    $0(A::One, B::One) => {}
+                    (A::One, B::Two) => {}
+                    (A::Two, B::Two) => {}
+                }
+            }
+            "#,
+        );
+    }
```
Comment (matklad) [inline crates/ide_assists/src/handlers/fill_match_arms.rs:511]: If we are touching this test anyway, lets unindent it as per our style guide.

**Review by matklad — COMMENTED** [review]
LGTM!

bors d+

moving this to diagnostics fix is something we should look into near/mid term

File: crates/ide_assists/src/handlers/fill_match_arms.rs (hunk 1 above)
Comment (iDawer) [reply crates/ide_assists/src/handlers/fill_match_arms.rs:511]: Done.

**Pull request merged**